test/functional/forall/test-forall-seq.cpp

```diff
@@ -0,0 +1,21 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA/RAJA.hpp"
+#include "gtest/gtest.h"
+
+#include "test-forall-rangesegment.hpp"
+
+// Generate Sequential Type List
+using SequentialTypes = list< RAJA::seq_exec, 
+                              RAJA::loop_exec,
+                              RAJA::simd_exec
+                            >;
+
+using SequentialForallTypes = Test<cartesian_product< IdxTypes, ListHost, SequentialTypes >>::Types;
```

Comment:
I didn't know cartesian_product existed. This makes enumeration much cleaner; I'll refactor my tests to use it. 👍

Reply:
FWIW, before this it didn't.  @mdavis36 asked for a way to do something like that, and I'd had it on my to-do list for a while, so we have it now.  Expansion is probably a bit slow for now, but it's there.